sql/postgres/sqlspec.go

```diff
@@ -63,7 +63,30 @@ func UnmarshalSpec(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{
 
 // MarshalSpec marshals v into an Atlas DDL document using a schemaspec.Marshaler.
 func MarshalSpec(v interface{}, marshaler schemaspec.Marshaler) ([]byte, error) {
-	return specutil.Marshal(v, marshaler, schemaSpec)
+	d := &doc{}
+	switch s := v.(type) {
+	case *schema.Schema:
+		spec, tables, enums, err := schemaSpec(s)
```

Comment:
I am not a fan of this API extending the number of returned values.
maybe we should return (struct,err) ?

Reply:
schemaSpec is unimported and therefore I think this is okay. If it had been an exported function, I'd agree with you. Let's see what @a8m thinks :-)